src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/GraphNodes/DependenciesGraphProvider.cs

```diff
@@ -104,10 +104,7 @@ protected override Task DisposeCoreAsync(bool initialized)
         /// </summary>
         public void BeginGetGraphData(IGraphContext context)
         {
-            ThreadHelper.JoinableTaskFactory.RunAsync(async () =>
-            {
-                await BeginGetGraphDataAsync(context);
-            });
+            ThreadHelper.JoinableTaskFactory.RunAsync(() => BeginGetGraphDataAsync(context));
```

Comment:
This reminds me, I saw [this guidance](https://github.com/davidfowl/AspNetCoreDiagnosticScenarios/blob/master/AsyncGuidance.md#prefer-asyncawait-over-directly-returning-task) the other day and thought it was interesting after our discussions about this in the past.

For the record I support this type of change :)

Reply:
Thanks for the link, I hadn't seen that.

We have a lot of `return Task.CompletedTask` throughout the code and CPS which goes against this guidance.